android/src/main/java/com/twiliorn/library/CustomVideoProcessor.java

```diff
@@ -89,7 +87,7 @@ public void onCapturerStopped() {
 
     @Override
     public void setSink(@Nullable VideoSink videoSink) {
-        // noop
```

Comment:
If I understand right, isn't the "sink" the output where the frame goes after this processor stage? I guess I'm confused with this as a noop, how did it ever work before?

Reply:
Previously I was using a custom `setVideoView` which was essentially the same as this `setSink`. Thats why I like I said in Slack, Idk really know why this worked and the previous implementation didn't. There really in my mind isnt much difference between the custom `setVideoView` i had and this `setSink` override. But yeah the sink/videoview is where the video is rendered

Reply:
Isn't it necessary though to both render the captured frame to the preview view, and to transmit it to the remote? Is that the difference between the sink and the videoView?  So maybe the video preview showed locally before but didn't actually get sent to the remote participant.  Or does the VideoProcessor pipeline work such that the VideoProcessor is only feeding the view (and Twilio handles sending it to the remote participant separately?)

Reply:
Nevermind, I guess that doesn't make any sense. I was also looking at some of their sample code and they are not even setting the sink.

Reply:
I basically got this code from their sample in my original implementation and that is why i didnt use the sink either. this overall really confused me lol. one of the weirder things in my dev career. I could probably find a more clear answer scouring through twilio source code but save that for another day lol

Reply:
Here is the example I used for future reference where Twilio themselves did not even override `setSink`. They instead used a constructor to pass in the `videoView` similar to what I originally did

https://github.com/twilio/video-quickstart-android/blob/master/exampleAdvancedCameraCapturer/src/main/java/com/twilio/video/examples/advancedcameracapturer/Photographer.kt

All I can say is this all just puzzled me

Reply:
I suspect the `setSink()` method is being called internally by Twilio with the correct sink for getting the frames to the remote end. So when this method was a noop, the correct sink was never being set. Obviously Twilio's SDK doesn't know anything about the custom `setVideoView()` method you had earlier, so it wouldn't have been calling it. I can't answer why this worked before rebasing on `main` though... Perhaps the behavior within the Twilio SDK changed in a more recent version?

For reference, this is what one version of the `VideoSource.setVideoProcessor()` method looks like:
https://chromium.googlesource.com/external/webrtc/+/HEAD/sdk/android/api/org/webrtc/VideoSource.java#128. Twilio's SDK internally uses a fork of the Android WebRTC library, so I can't say if this matches the implementation within the Twilio SDK. But you can at least see that in this version, it does call the `VideoProcessor.setSink()` method as part of configuring the passed-in `VideoProcessor`.